### PR TITLE
feature/BOT-3322-coach-coach-worker---use-redis-

### PR DESCRIPTION
**Pull request by murliwatz**

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -23,36 +22,60 @@
...
 import inspect
+import redis
+from api.utils.log import getLogger
 
-def process_scheduler(req_queue,log_format,log_level,log_datefmt):
-    logger = logging.getLogger('Worker scheduler')
-    logger.setLevel(log_level)
+def redis_scheduler(req_queue):
+    in_queue = []
+    red = redis.Redis(host=os.environ.get('REDIS_HOST', 'localhost'), port=6379, db=os.environ.get('REDIS_DB', 0))
```
Comment (codeforequity-at) [inline main.py:30]: will this work for clustered Redis ?

**Review by codeforequity-at — CHANGES_REQUESTED** [review]

File: main.py [hunk 2]
```diff
@@ -23,36 +22,60 @@
...
+        for k in red.scan_iter("coachworker_req*"):
+            if k not in in_queue:
+                if k.decode("utf-8").startswith('coachworker_req_chi2'):
+                    req_obj = json.loads(red.get(k))
+                    req_queue.put((req_obj, "calculate_chi2"))
+                    red.set('coachworker_status_chi2_' + req_obj['coachSessionId'], json.dumps({
+                        "method": "calculate_chi2",
+                        "clientId": req_obj['clientId'],
+                        "coachSessionId": req_obj['coachSessionId'],
+                        "status": ['Request for for chi2 analysis is queued']
```
Comment (codeforequity-at) [inline main.py:41]: Can we use "status" as a code/enum and place this in an additional field "statusDescription"
Also, an additional status can be "calculations running", maybe even with some kind of progress - not as percentage, but maybe as "step 3/5 completed ..."

File: main.py [hunk 3]
```diff
@@ -23,36 +22,60 @@
...
 import inspect
+import redis
+from api.utils.log import getLogger
 
-def process_scheduler(req_queue,log_format,log_level,log_datefmt):
-    logger = logging.getLogger('Worker scheduler')
-    logger.setLevel(log_level)
+def redis_scheduler(req_queue):
+    in_queue = []
+    red = redis.Redis(host=os.environ.get('REDIS_HOST', 'localhost'), port=6379, db=0)
```
[the remark this answers is not in the document]
Comment (murliwatz) [reply main.py]: done

File: main.py (hunk 1 above)
Comment (murliwatz) [reply main.py]: added sentinel support

File: main.py (hunk 2 above)
Comment (murliwatz) [reply main.py]: changed status update functionality

File: main.py [hunk 4]
```diff
@@ -23,36 +22,60 @@
...
+def redis_scheduler(req_queue):
+    in_queue = []
+    red = redis.Redis(host=os.environ.get('REDIS_HOST', 'localhost'), port=6379, db=os.environ.get('REDIS_DB', 0)) 
+    while True:
+        for k in red.scan_iter("coachworker_req*"):
+            if k not in in_queue:
+                if k.decode("utf-8").startswith('coachworker_req_chi2'):
+                    req_obj = json.loads(red.get(k))
+                    req_queue.put((req_obj, "calculate_chi2"))
+                    red.set('coachworker_status_chi2_' + req_obj['coachSessionId'], json.dumps({
```
[the remark this answers is not in the document]
Comment (murliwatz) [reply main.py:39]: redis keys expire now

**Pull request merged**